src/backend/logger/logger.ts

```diff
@@ -514,12 +516,31 @@ const shouldToggleShaderPreCacheOn = async (
     LogPrefix.Backend
   )
 
-  return [
+  const envVarsNotSet = [
     'STEAM_COMPAT_TRANSCODED_MEDIA_PATH',
     'STEAM_COMPAT_MEDIA_PATH',
     'STEAM_FOSSILIZE_DUMP_PATH',
     'DXVK_STATE_CACHE_PATH'
   ].every((envVar) => !process.env[envVar])
+
+  if (envVarsNotSet) {
+    await dialog.showMessageBox({
+      type: 'warning',
+      title: i18next.t(
+        'box.shaderPreCachingDisabledTitle',
+        'Shader Pre-Caching Disabled'
+      ),
+      message: i18next.t(
+        'box.shaderPreCachingDisabledMessage',
+        "Steam's Shader Pre-cache is disabled. Please enable it on the Steam Settings in Desktop Mode to ensure the game works properly with UMU."
+      ),
+      buttons: [i18next.t('box.launchAnyway', 'Launch Game Anyway')]
+    })
+
+    return true
+  }
```

Comment:
I'd move this to somewhere in `launchEventCallback`. This function's name suggests that it only *checks* whether the option is toggled, actually doing something based on that status should be left to somewhere else